ext/DynamicHMCExt.jl

```diff
@@ -0,0 +1,95 @@
+module DynamicHMCExt
+###
+### DynamicHMC backend - https://github.com/tpapp/DynamicHMC.jl
+###
+
+
+if isdefined(Base, :get_extension)
+    import DynamicHMC
+    using Turing
+    using Turing: AbstractMCMC, Random, LogDensityProblems 
+    using Turing.Inference: LogDensityProblemsAD
+else
+    import ..DynamicHMC
+    using ..Turing
+    using ..Turing: AbstractMCMC, Random, LogDensityProblems 
+    using ..Turing.Inference: LogDensityProblemsAD
+end
+
+# Wraps DynamicHMC as an AbstractSampler
+struct DynamicNUTS{S<:DynamicHMC.NUTS} <: AbstractMCMC.AbstractSampler
+    sampler::S
+end
+
+DynamicNUTS() = DynamicNUTS(DynamicHMC.NUTS())
+Turing.externalsampler(spl::DynamicHMC.NUTS) = Turing.externalsampler(DynamicNUTS(spl))
+
+"""
+    DynamicNUTSState
+
+State of the [`DynamicNUTS`](@ref) sampler.
+"""
+struct DynamicNUTSState{L,C,M,S}
+    logdensity::L
+    "Cache of sample, log density, and gradient of log density evaluation."
+    cache::C
+    metric::M
+    stepsize::S
+end
+
+function AbstractMCMC.step(
+    rng::Random.AbstractRNG,
+    model::AbstractMCMC.LogDensityModel,
+    spl::DynamicNUTS;
+    init_params = nothing,
+    kwargs...
+)
+    # Unpack model
+    # We wrap it again in ADgradient 
+    ℓ = LogDensityProblemsAD.ADgradient(model.logdensity)
+
+    # Make init params if nothing
+    if init_params == nothing
```

Comment:
```suggestion
    if init_params === nothing
```
is generally preferrable.